test/handlers/checksum_test.py

```diff
@@ -35,7 +34,7 @@
     ids=str
 )
 def user_file(request):
-    with storage.Backend(request.param) as backend:
+    with request.param as backend:
         yield backend
```

Comment:
Since `userstorage.Base.__enter__()` returns self, we don't really need to get the
return value of the context manager, so we can simply do:

    with request.param:
        yield request.param

Reply:
This is technically the same as current code? Just different ways of aliasing `request.param`, no?
```
backend = request.param
with backend:
    yield backend
```

Reply:
Current code yields the the result of the with statement, which is the same since we backend.__enter__() returns self.

Both options work, my suggestion was to use the simplest possible form.

Reply:
I understand why they result in the same behavior, but I do not really see how it is simpler or better assigning the parameter before the with statement or using the result of the with-as statement. Different flavors of the same code.

Using the result of the with statement has the advantage of:
- with-as is a recognizable statement
- if in the future userstorage changes return in `__enter__` (unlikely...), we get the new returned object already by updating
- is already coded like that in the current version!

Reply:
The current code is confusing:

    with request.param as backend:

This looks like request.param is not a backend and the result of the with statement is a backend. This confuses the reader that there is something going on under the hood while there is no such thing.

The suggested code:

    with request.param:
        yield request.param

I clear that we must yield request.param in the context of a with statement. But this does not reveal what is request.param, this is why I suggest the option to have an explaining variable:

     backend = request.param

Just to make it clear that the fixture is accepting a backend as parameter.

There is also another issue caused by using the result of the with statement. In the case when we request.param is a tuple, we want to unpack it:

    backend, can_detect_sector_size = request.param

In the current code you invented a backend_file (is this the file of a backend?) so we can use the name backend for the result of the with statement. If we don't use the result of the with statement we don't need to invent new names.

Usually context manager return self. In some cases like tempfile.TemporaryDirectory this is not the case and the result of the with statement is a string instead of the temporary directory object. This is unexpected and confusing.

Future version of userstorage cannot return different value from the with statement since this will be backward incompatible with current users of the code not using the result of the with statement.

Reply:
I understand the change for the cases that require the tuple, I will do the change.

For this case it feels like saying that this is confusing:
```
with open('file', 'r') as f:
     f.read()
```
And is clearer to do:
```
f = open('file', 'r')
with f:
    f.read()
```
I can see what got you there with what you explained. But not sure, I do not see it confusing. Maybe a little obscure as you can only assume what the `__enter__` is returning, but the same happens in practice every time we use a context. In general, you just expect it to behave.

But anyway, if you really think this is clearer, I'll do the change tomorrow.

Reply:
The open example is not the same. Calling open() returns an open file object that can be used as context manager to ensure the file is closed. Creating a temporary variable is typically a bad idea since it can be broken by adding another line after creating the file. If the other line raises, we will not close the open file. `f` is the common name for an open file, so everything is clear.

In our case we have a request.param, which is already a backend, but this is not clear when you read the code. We need to use a with statement with the backend and yield it to the caller - nothing else is needed. The suggested code does exactly this.

This is little detail, but we are going to repeat this code in 2 projects, so I think it worth the time to do this change in the best way.

Reply:
True, both are already a backend, but one of them is `setup()`, and will be `teardown()` at the end. We could maybe use `backend`, or `b` to signify that a temporary variable is a setup backend.

Reply:
I don't think we need a different name for a setup backend. Inside the context, the backend is set up, and outside the context it is not.